pkg/util/util_test.go

```diff
@@ -0,0 +1,264 @@
+							"includeIPRanges":              "*",
+							"autoInject":                   "enabled",
+							"tracer":                       "zipkin",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	icpYaml = `
+defaultNamespacePrefix: istio-system
+hub: docker.io/istio
+tag: 1.1.4
+defaultNamespacePrefix: istio-system
+profile: default
+
+trafficManagement:
+  enabled: true
+  components:
+    namespace: istio-control
+    pilot:
+      common:
+        k8s:
+          env:
+          - name: GODEBUG
+            value: "gctrace=1"
+          hpaSpec:
+            maxReplicas: 5
+            minReplicas: 1
+            scaleTargetRef:
+              apiVersion: apps/v1
+              kind: Deployment
+              name: istio-pilot
+            metrics:
+              - type: Resource
+                resource:
+                  name: cpu
+                  targetAverageUtilization: 80
+          replicaCount: 1
+          readinessProbe:
+            handler: {}
+# TODO: uncomment the following lines after the jsonpb unmarshaling issue is resolved: https://github.com/istio/istio/issues/15366
+#            httpGet:
+#              path: /ready
+#              port: 8080
+            initialDelaySeconds: 5
+            periodSeconds: 30
+            timeoutSeconds: 5
+          resources:
+            requests:
+              cpu: 500m
+              memory: 2048Mi
+        values:
+          image: pilot
+          traceSampling: 1.0
+          configNamespace: istio-config
+          keepaliveMaxServerConnectionAge: 30m
+          configMap: true
+          ingress:
+            ingressService: istio-ingressgateway
+            ingressControllerMode: "OFF"
+            ingressClass: istio
+          telemetry:
+            enabled: true
+          policy:
+            enabled: false
+          useMCP: true
+    proxy:
+      common:
+        values:
+          image: proxyv2
+          clusterDomain: "cluster.local"
+          resources:
+            requests:
+              cpu: 100m
+              memory: 128Mi
+            limits:
+              cpu: 2000m
+              memory: 128Mi
+          concurrency: 2
+          accessLogEncoding: TEXT
+          logLevel: warning
+          componentLogLevel: "misc:error"
+          dnsRefreshRate: 300s
+          privileged: false
+          enableCoreDump: false
+          statusPort: 15020
+          readinessInitialDelaySeconds: 1
+          readinessPeriodSeconds: 2
+          readinessFailureThreshold: 30
+          includeIPRanges: "*"
+          autoInject: enabled
+          tracer: "zipkin"
+`
+)
+
+func TestToYAMLWithJSONPB(t *testing.T) {
+	toYAMLWithJSONPBTests := []struct {
+		desc string
+		pb   *v1alpha2.IstioControlPlaneSpec
+		want string
+	}{
+		{"TranslateICPToYAMLWithJSONPB", icp, icpYaml},
+	}
+
+	for _, tt := range toYAMLWithJSONPBTests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := ToYAMLWithJSONPB(icp)
+			if !IsYAMLEqual(got, tt.want) {
```

Comment:
As an extra test in here I suggest adding || YAMLDiff != ""

Reply:
done